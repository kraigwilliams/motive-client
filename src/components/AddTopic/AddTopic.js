import React, { Component } from "react";
import { FormButton, GoBack } from "../Button/Button";
import {
  FormWrapper,
  FormTitle,
  FormLabel,
  FormInput,
  Required,
} from "../Form/Form";
import { PageWrapper, colors } from "../constants";
import ContentService from "../../services/content-service";

export default class AddTopic extends Component {
  handleSubmit = (ev) => {
    ev.preventDefault();
    const { title, description } = ev.target;
    ContentService.postTopic(title.value, description.value).then((topic) => {
      const topicId = topic.id;
      this.props.history.push(`topics/${topicId}`);
    });
  };

  render() {
    document.title = "Add Topic - Folkul";
    return (
      <PageWrapper padding="40px 0" bgcolor={colors.darkgrey}>
        <FormWrapper
          padding="50px"
          onSubmit={this.handleSubmit}
          bgcolor={colors.darkgrey}
        >
          <GoBack
            type="reset"
            onClick={() => this.props.history.goBack()}
            margin="0px"
            color={colors.slategrey}
          />

          <FormTitle>Create a Topic</FormTitle>

          <FormLabel htmlFor="topic-title">
            Title <Required />
          </FormLabel>
          <FormInput
            id="topic-title"
            name="title"
            aria-label="Enter title of this topic"
            aria-required="true"
            required
            color={colors.white}
            placeholder="Walden Pond"
          />

          <FormLabel htmlFor="topic-description">Description</FormLabel>
          <FormInput
            id="topic-description"
            name="description"
            aria-label="Enter description of this topic"
            color={colors.white}
            placeholder="“Things do not change; we change.”"
          />

          {/* <FormLabel htmlFor='topic-thoughts'>
            Add Thoughts to Topic
          </FormLabel>
          <Dropdown 
            id='topic-thoughts'
            name='thoughts'
            aria-label="Select any exisiting thoughts that belong within this topic"
          />

          <FormLabel htmlFor='topic-connects'>
            Add Connections to Topic
          </FormLabel>
          <Dropdown 
            id='topic-connects'
            name='connects'
            aria-label="Select any connections you wish to share this topic with"
          /> */}

          <FormButton type="submit" color={colors.coral}>
            Submit
          </FormButton>
        </FormWrapper>
      </PageWrapper>
    );
  }
}
