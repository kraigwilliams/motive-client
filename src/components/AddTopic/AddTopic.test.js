import React from 'react'
import ReactDOM from 'react-dom'
import AddTopic from './AddTopic'
import renderer from 'react-test-renderer';
    
describe(`<AddTopic />`, () => {
  //Smoke Testing 
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(
      <AddTopic />,
      div
    )
    ReactDOM.unmountComponentAtNode(div)
  })
  //Snapshot Testing
  it('renders UI as expected', () => {
    const tree = renderer 
      .create(<AddTopic />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})