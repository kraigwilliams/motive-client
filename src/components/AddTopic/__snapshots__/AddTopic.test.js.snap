// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddTopic /> renders UI as expected 1`] = `
<div
  className="sc-AxjAm dtabnj"
>
  <form
    className="sc-fzoLsD eWscXS"
    onSubmit={[Function]}
  >
    <button
      className="Button sc-AxiKw duGoGH"
      color="#829191"
      margin="0px"
      onClick={[Function]}
      type="reset"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-arrow-left fa-w-14 "
        data-icon="arrow-left"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M257.5 445.1l-22.2 22.2c-9.4 9.4-24.6 9.4-33.9 0L7 273c-9.4-9.4-9.4-24.6 0-33.9L201.4 44.7c9.4-9.4 24.6-9.4 33.9 0l22.2 22.2c9.5 9.5 9.3 25-.4 34.3L136.6 216H424c13.3 0 24 10.7 24 24v32c0 13.3-10.7 24-24 24H136.6l120.5 114.8c9.8 9.3 10 24.8.4 34.3z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </button>
    <h1
      className="sc-fzozJi gnsJxi"
    >
      Create a Topic
    </h1>
    <label
      className="Label sc-fzplWN iwXlQR"
      htmlFor="topic-title"
    >
      Title 
      <span
        className="Required"
      >
        *
      </span>
    </label>
    <input
      aria-label="Enter title of this topic"
      aria-required="true"
      className="Input sc-fznyAO iAKbbp"
      color="#ffffff"
      id="topic-title"
      name="title"
      required={true}
      type="text"
    />
    <label
      className="Label sc-fzplWN iwXlQR"
      htmlFor="topic-description"
    >
      Description
    </label>
    <input
      aria-label="Enter description of this topic"
      className="Input sc-fznyAO iAKbbp"
      color="#ffffff"
      id="topic-description"
      name="description"
      type="text"
    />
    <button
      className="Button sc-AxhUy jPkvRe"
      color="#EB9C82"
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;
